CMakeLists.txt

```diff
@@ -335,13 +352,39 @@ else()
       -DWEBRTC_LINUX
     )
 
-    target_compile_options(${MODULE} PRIVATE
-      -B${libwebrtc_source_dir}/src/third_party/binutils/Linux_x64/Release/bin
-      -fpermissive
-      -nostdinc++
-      -nodefaultlibs
-      --sysroot=${libwebrtc_source_dir}/src/build/linux/debian_sid_amd64-sysroot
-    )
+    if ("$ENV{TARGET_ARCH}" STREQUAL "arm")
+      target_compile_options(${MODULE} PRIVATE        
+        -fpermissive
+        -nostdinc++
+        -nodefaultlibs
+        -march=armv7-a
+        -mfloat-abi=hard
+        -mtune=generic-armv7-a
+        -mfpu=neon
+      )
+
+      set(CMAKE_SYSTEM_NAME Linux)
+      set(CMAKE_SYSTEM_PROCESSOR arm)
+      
+      set(CMAKE_SYSROOT ${libwebrtc_source_dir}/src/build/linux/debian_sid_arm-sysroot)
+
+      set(tools $ENV{ARM_TOOLS_PATH})
+      set(CMAKE_C_COMPILER ${tools}/bin/arm-linux-gnueabihf-gcc)
+      set(CMAKE_CXX_COMPILER ${tools}/bin/arm-linux-gnueabihf-g++)
+
+      set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+      set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+      set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+      set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+    else()
+      target_compile_options(${MODULE} PRIVATE
+        -B${libwebrtc_source_dir}/src/third_party/binutils/Linux_x64/Release/bin
+        -fpermissive
+        -nostdinc++
+        -nodefaultlibs
```

Comment:
Like above, can you factor out the common args? For example,

* `-fpermissive`
* `-nostdinc++`
* `-nodefaultlibs`

Reply:
OK